### PR TITLE
Merge updates from ocm-plus repo

### PR DESCRIPTION
**Pull request by JustinKuli**
Brings in commits up to the current "release-2.4" branch from the other repo.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -142,12 +146,22 @@ func main() {
 				os.Exit(1)
 			}
 		} else {
+
 			log.Info("Starting lease controller to report status")
 			leaseUpdater := lease.NewLeaseUpdater(
 				generatedClient,
-				"config-policy-controller",
+				"policy-controller",
```
Comment (dhaiducek) [inline cmd/manager/main.go:153]: This change was reverted in
- https://github.com/open-cluster-management/config-policy-controller/pull/175

So maybe pull some more PRs from midstream in? I'm not 100% sure of the effect here, but keeping `config-policy-controller` seems clearer to me...

**Review by dhaiducek — COMMENTED** [review]
Wow! Thanks for tackling this! I looked pretty quickly and noted one thing that stuck out. I'm hoping the tests are capturing anything I might have missed.

File: cmd/manager/main.go (hunk 1 above)
Comment (JustinKuli) [reply cmd/manager/main.go:153]: Nice catch

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**